[KOGITO-858] Add a Decision Table RuleUnit example

tkobayas opened the pull request:
decisiontable-quarkus-example and decisiontable-springboot-example.

Please review, thanks!

evacchi commented on decisiontable-quarkus-example/src/test/java/org/kie/kogito/QueryTest.java:
```diff
@@ -0,0 +1,84 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class QueryTest {
```
it's probably better for now to remove these tests against the programmatic APIs, as 1) they duplicate the http tests 2) they test an API that is still not stable nor public 3) they pose some problems when we use the quarkus extension instead of the maven plugin (basically they break the build because unfortunately the extension runs *after* the tests, in the package phase...)

MarianMacik replied:
So that is the reason why we use `kogito-maven-plugin` also with Quarkus examples instead of Kogito Quarkus extension?

evacchi replied:
no, that's a side effect. 

the reason why we use the maven plugin is because it was for hard to use the latest codegen (SNAPSHOT) because the quarkus bom controlled versioning of the extension (and therefore of the runtimes). If we wanted to use the extension, we would have to redeclare all dependencies explicitly in all poms. Now that we can control it, we can provide our own "kogito-quarkus-bom" pointing to SNAPSHOT

tkobayas replied:
Removed QueryTest. Thanks!

evacchi commented on decisiontable-springboot-example/src/test/java/org/kie/kogito/QueryTest.java:
```diff
@@ -0,0 +1,84 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class QueryTest {
```
let's remove it from here too, to avoid confusion

MarianMacik replied:
+1 for consistency

tkobayas replied:
Removed QueryTest. Thanks!

tkobayas commented:
Thank you for reviewing!

I have updated the PR:
- Use quarkus extension instead of kogito-maven-plugin
- Removed QueryTest from decisiontable-quarkus-example and decisiontable-springboot-example

evacchi commented on decisiontable-quarkus-example/pom.xml:
```diff
@@ -0,0 +1,137 @@
+      </plugin>
+      <plugin>
+        <groupId>io.openapitools.swagger</groupId>
+        <artifactId>swagger-maven-plugin</artifactId>
```
you should be able to get rid of this plugin and use `quarkus-smallrye-openapi` instead :)

tkobayas replied:
Good to know. Thanks!

tkobayas commented:
Update the PR:
- Removed swagger-maven-plugin from decisiontable-quarkus-example
